addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner.js

```diff
@@ -0,0 +1,137 @@
+                          /ODBC DB2/igm,
+                          /ODBC Driver/igm,
+                          /ODBC Error/igm,
+                          /ODBC Oracle/igm,
+                          /OLE\/DB provider returned message/igm,
+                          /Oracle DB2/igm,
+                          /Oracle Driver/igm,
+                          /Oracle Error/igm,
+                          /Oracle ODBC/igm,
+                          /PHP Error/igm,
+                          /PHP Parse error/igm,
+                          /PHP Warning/igm,
+                          /Parent Directory/igm,
+                          /Permission denied: 'GetObject'/igm,
+                          /PostgreSQL query failed: ERROR: parser: parse error/igm,
+                          /The script whose uid is/igm,
+                          /Type mismatch/igm,
+                          /Unable to jump to row/igm,
+                          /Unterminated string constant/igm,
+                          /Warning: Cannot modify header information - headers already sent/igm,
+                          /Warning: Supplied argument is not a valid File-Handle resource in/igm,
+                          /Warning: mysql_query\(\)/igm,
+                          /Warning: pg_connect\(\): Unable to connect to PostgreSQL server: FATAL/igm,
+                          /data source=/igm,
+                          /invalid query/igm,
+                          /is not allowed to access/igm,
+                          /mySQL error with query/igm,
+                          /on MySQL result index/igm,
+                          /server object error/igm];
+
+function scan(helper,msg) {
+
+    if(msg.opcode != OPCODE_TEXT || msg.isOutgoing){
+        return;
+    }
+
+    var counter = 0;
+    var message = String(msg.getReadablePayload()).valueOf();
+
+    strings_msql_error.forEach(function(pattern){
+        if((matches = message.match(pattern)) != null){
+            matches.forEach(function(evidence){
+
+                helper.newAlert()
+                    .setRiskConfidence(RISK_MEDIUM, CONFIDENCE_MEDIUM)
+                    .setName("Application Error Disclosure via WebSockets (script)")
+                    .setDescription(
+                        "This page contains an error/warning message that \
+ may disclose sensitive information like the location of the file \
+ that produced the unhand-led exception. This information can be used\
+ to launch further attacks against the web application. The alert\
+ could be a false positive if the error message is found inside a\
+ documentation page.")
+                    .setSolution("Review the source code of this page. Implement custom \
+ error pages. Consider implementing a mechanism to provide a unique\
+ error reference/identifier to the client (browser) while logging the\
+ details on the server side and not exposing them to the user.")
+                    .setParam(counter++)
```

Comment:
I think this is probably used on some handling in the [Find HTML Comments](https://github.com/zaproxy/community-scripts/blob/27efdf1cb39dee4f9dd71d7055e4b5a6ee231c42/passive/Find%20HTML%20Comments.js#L23-L24) script. However: 1) It's an older script. 2) It uses parameters in order to facilitate different grouping of results.

In the majority of cases passive scanners (scripts or other) should not really have/need a "parameter".

Reply:
Fixed in 319f60c